TextReader.java

```diff
@@ -1 +1,79 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
 
+public class TextReader {
+
+	private ArrayList<String> lines = new ArrayList<String>();
+	Map<String, ArrayList<Integer>> wordMap = new HashMap<>();
+
+	public void read(String fileName, String pattern) {
+
+		setUpWordMap(fileName);
+
+		Set<Integer> matchedLines = matchWords(wordMap, pattern);
+
+		printResults(matchedLines,lines);
+	}
+
+
+	private void setUpWordMap(String fileName) {
+		try {
+			File file = new File(fileName);
+			Scanner scanner = new Scanner(file);
+			while (scanner.hasNextLine()) {
+				String data = scanner.nextLine();
+				lines.add(data);
+			}
+	
+			// adds all words in each line into wordList
+			int index = 0;
+			for (String line : lines) {
+				for (String word : line.split("\\s+")) {
+					if (!wordMap.containsKey(word)) {
+						wordMap.put(word, new ArrayList<>());
+						wordMap.get(word).add(index);
+					} else {
+						wordMap.get(word).add(index);
+					}
+				}
+				index++;
+			}
```

Comment:
Searching for "sister" gives no result, but searching for "sister." does, we suggest stripping the raw word after the split of all non-letter characters.
```suggestion
                        // adds all words in each line into wordList
                        int index = 0;
                        for (String line : lines) {
                            for (String rawWord : line.split("\\s+")) {
                                String word = rawWord.replaceAll("[^a-zA-Z]", "");
                                if (!wordMap.containsKey(word)) {
                                    wordMap.put(word.replaceAll("[^a-zA-Z]", ""), new ArrayList<>());
                                    wordMap.get(word).add(index);
                                } else {
                                    wordMap.get(word).add(index);
                                }
                            }
                            index++;
                        }
```